src/processing.ts

```diff
@@ -44,3 +47,26 @@ export function assignParents(node: Node): void {
         assignParents(c);
     }
 }
+
+Node.prototype.hasValidParents = function(parent?: Node) : boolean {
+    return this.parent == parent &&
+        this.children
+            .map(c => c.hasValidParents(this))
+            .reduce((res, val) => res && val, true);
+}
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+function* getInvalidPositions(node: Node): Generator<Node> {
+    for (const n of this.walk()) {
+        if (n.position == undefined || (
+            // If the parent position is null, we can't say anything about this node's position
+            n.parent?.position != undefined && !(n.parent?.position.contains(n.position.start) && n.parent?.position.contains(n.position.end))
+        )) {
+            yield n;
+        }
+    }
+}
+
+Node.prototype.invalidPositions = function() {
+    return getInvalidPositions(this);
+}
```

Comment:
Great that you've also added these, I think there are tests ported from Kolasu that used these checks and have been removed